modules/markup/markdown/goldmark.go

```diff
@@ -79,6 +79,9 @@ func (g *GiteaASTTransformer) Transform(node *ast.Document, reader text.Reader,
 				}
 				link = []byte(giteautil.URLJoin(pc.Get(urlPrefixKey).(string), lnk))
 			}
+			if len(link) > 0 && link[0] == '#' {
```

Comment:
The `if` can be rearranged with the code above so the conditions are only checked once (e.g. `len(link)>0` and `link[0]=='#'`)

Reply:
I don't know if this is more readable so I would like some other opinions whether this is worth it.

Reply:
@guillep2k the first `if` has `link[0] != '#'` and the second has `link[0] == '#'` (note: the `!` in the first that isn't in second), so I think that even though there is one conditional in the `if` that overlaps, any computation time that would be saved is minimal, and potentially even make the whole section of code be less readable (but that is from my rough attempts to combine them which I haven't spent too much time trying, I'd be happy to be proven wrong).